metadata/lib.rs

```diff
@@ -28,16 +28,17 @@
 //! let language = SourceLanguage::new(Language::Ink, Version::new(2, 1, 0));
 //! let compiler = SourceCompiler::new(Compiler::RustC, Version::parse("1.46.0-nightly").unwrap());
 //! let source = Source::new([0u8; 32], language, compiler);
-//! let contract = Contract::new(
-//!     "incrementer".to_string(),
-//!     Version::new(2, 1, 0),
-//!     vec!["Parity Technologies <admin@parity.io>".to_string()],
-//!     Some("increment a value".to_string()),
-//!     Some(Url::parse("http://!docs.rs/").unwrap()),
-//!     Some(Url::parse("http://!github.com/paritytech/ink/").unwrap()),
-//!     Some(Url::parse("http://!example.com/").unwrap()),
-//!     Some("Apache-2.0".to_string()),
-//! );
+//! let contract = Contract::builder()
+//!     .name("incrementer".to_string())
+//!     .version(Version::new(2, 1, 0))
+//!     .authors(vec!["Parity Technologies <admin@parity.io>".to_string()])
+//!     .description(Some("increment a value".to_string()))
+//!     .documentation(Some(Url::parse("http:docs.rs/").unwrap()))
+//!     .repository(Some(Url::parse("http:github.com/paritytech/ink/").unwrap()))
+//!     .homepage(Some(Url::parse("http:example.com/").unwrap()))
+//!     .license(Some("Apache-2.0".to_string()))
```

Comment:
I would expect from a builder pattern to receive the arguments as `T` and not `Option<T>` so if you just do not use `.description(whatever)` the `description` of a contract would implicitly be `None`.

Reply:
Yes I had it this way originally, however in the place where it is used https://github.com/paritytech/cargo-contract/blob/ea01a8a540faa3c718e84d1bf3003db09315b5fb/src/cmd/metadata.rs#L119-L129 the values I am passing are already `Option`s, so it required lots of `if let Some(opt) = opt { builder.opt(opt) }`.

Could have both by adding an extra method for each field e.g. `description_opt`?

Reply:
Done